util/metadata_cleanup.py

```diff
@@ -2,70 +2,115 @@
 from acq4.util.DataManager import getDirHandle
 from multipatch_analysis.experiment import Experiment
 
-csv_file = 'Slice_Preparation_Record.csv'
+slice_csv_file = 'Slice_Preparation_Record.csv'
+osm_csv_file = 'MultiPatch Throughput.xlsx - Solution Making.csv'
 
 # read all dissection times
-recs = list(csv.reader(open(csv_file).readlines()))
-id_col = recs[0].index('LabTracks ID')
-date_col = recs[0].index('Date Sliced')
-sac_col = recs[0].index('Time Mouse Sacrificed')
-time_col = recs[0].index('heated ACSF1 incub. Comp.')
-diss_times = {rec[id_col]: rec for rec in recs[1:]}
+recs = list(csv.reader(open(slice_csv_file).readlines()))
+diss_columns = {
+    'id_col': recs[0].index('LabTracks ID'),
+    'date_col': recs[0].index('Date Sliced'),
+    'sac_col': recs[0].index('Time Mouse Sacrificed'),
+    'time_col': recs[0].index('heated ACSF1 incub. Comp.')}
+diss_times = {rec[diss_columns['id_col']]: rec for rec in recs[1:]}
 
+# read solution making
+osm_recs = list(csv.reader(open(osm_csv_file).readlines()))
+osm_columns = {
+    'osm_date_col': osm_recs[0].index('Date'),
+    'recipe_col': osm_recs[0].index('Recipe'),
+    'osm_col': osm_recs[0].index('osmolarity')}
+osm_dates = {osm_rec[osm_columns['osm_date_col']]: osm_rec for osm_rec in osm_recs[1:]}
 
-root = sys.argv[1]
+def solution_check(dh, expt_date, columns):
+    solution = dh.info().get('solution', None)
+    osm = dh.info().get('solution_osm', None)
+    date = expt_date.strftime('%#m/%#d/%Y')
+    try:
+        osm_entry = osm_dates[date]
+        recipe = osm_entry[columns['recipe_col']]
+        sheet_osm = osm_entry[columns['osm_col']]
+        if solution is None:
+            print("\tSet aCSF: %s\r" % recipe)
+            return
 
-# find all subject folders that contain at least one site folder
-sites = glob.glob(os.path.join(root, '*', 'slice_*', 'site_*'))
-subs = sorted(list(set([os.path.abspath(os.path.join(s, '..', '..')) for s in sites])))
+        calcium = solution.split('m')[0]
+        if calcium in recipe:
+            print("\taCSF: Match!\r")
+        else:
+            print("\taCSF disagreement: %s != %s\r" % (solution, recipe))
 
-for path in subs:
-    dh = getDirHandle(path)
-    sub_id = dh.info().get('animal_ID', None)
+        if osm is None:
+            print("\tSet osmolarity: %s\r" % sheet_osm)
+            return
+
+        if sheet_osm == osm:
+            print("\tOsmolarity: Match!\r")
+        else:
+            print("\tOsmolarity disagreement: %s != %s\r" % (osm, sheet_osm))
+    except KeyError:
```

Comment:
This is dangerous--if a KeyError happens *anywhere* in the `try` block except the first line, then you get the wrong error message. Also big nested blocks are messy. Instead:

    try:
        osm_entry = osm_dates[date]
    except KeyError:
       print("\tSolution date not in Multipatch_throughput sheet, check notes\r")